### PR TITLE
chore: update hmi server api to align with v032 data service api

### LINKED ISSUE #290: [TASK]: update hmi-server api to align with v0.3.2 data-service api
**Describe the task**
Update HMI resources/proxies/models to be in line with v0.3.2 of the data service api

**Task Items**
- [ ] Align with data service 0.3.2
- [ ] Address feedback from @mwdchang on #261 about enum case sensitivity

**Goal**
A clear and concise description of what the end result must have for it to be approved as complete.

**Screenshots**
If applicable, add screenshots or designs

**Additional context**
Add any other context about the task here

### PR DESCRIPTION
**Pull request by dvince2**
# Description

Updating the API to align with data service v0.3.2. 
Addressing enum issue in our API

Resolves #290

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide reproduction instructions and list any relevant details for your test configuration

- Ran data service with terarium and manually tested endpoints.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by YohannParis — APPROVED** [review]

**Review by YohannParis — APPROVED** [review]

File: packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/dataservice/ProjectResource.java [hunk 1]
```diff
@@ -80,7 +79,8 @@ public Response createAsset(
 		@PathParam("resource_type") final String type, // ResourceType
 		@PathParam("resource_id") final String resourceId
 	) {
-		return proxy.createAsset(projectId, type, resourceId);
+		// String#toUpperCase is being used here to correct for Enum#valueOf usage, which we can't override
+		return proxy.createAsset(projectId, type.toUpperCase(), resourceId);
```
Comment (dgauldie) [inline packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/dataservice/ProjectResource.java:83]: this assumes that the value of the enum is just the lower case of the key. This is not very robust. I would suggest adding a look up function to the enum like:

```
public static ResourceType findByType(final String type) {
		return Arrays.stream(values()).filter(
			value -> value.type.equals(type)).findFirst().orElseThrow(() -> new IllegalArgumentException("No ResourceType with type: " + type)
		);
	}
```

and then using something like `ResourceType.findByType(type).type`

File: packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/dataservice/ProjectResource.java [hunk 2]
```diff
@@ -90,7 +90,7 @@ public Response deleteAsset(
 		@PathParam("resource_type") final String type, // ResourceType
 		@PathParam("resource_id") final String resourceId
 	) {
-		// String#toUpperCase is being used here to correct for Enum#valueOf usage, which we can't override
-		return proxy.deleteAsset(projectId, type.toUpperCase(), resourceId);
+		System.out.println(ResourceType.findByType(type));
```
Comment (YohannParis) [inline packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/dataservice/ProjectResource.java:93]: Should we keep comments out of PRs?

Comment (dvince2) [reply packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/dataservice/ProjectResource.java:93]: oh jeez... thanks!

**Review by YohannParis — APPROVED** [review]
LGTM

**Review by dgauldie — APPROVED** [review]
LGTM

**Pull request merged**